Add filler items

Darrman opened the pull request:
Intended to add extra filler items into the item pool to fill in empty locations.
Code doesn't currently work, especially the enums.

Darrman committed:
Unfinished filler item addition

The enums don't work properly and have ranges all over the place.

CT075 commented on bin/connector_config/Generate.hs:
```diff
@@ -37,38 +37,66 @@ data HolyWeapon
     | C5x
     | Endgame
     | Victory
+    | R10
```
We don't really need Ruins 10 to be its own location. 5x is just split out because "5x" isn't a number.

CT075 commented on bin/connector_config/Generate.hs:
```diff
@@ -37,38 +37,66 @@ data HolyWeapon
 instance Bounded Chapter where
     minBound = Prologue
-    maxBound = Victory
+    maxBound = R10
```
Organizationally, I think we should keep Victory as the last chapter, or remove Victory altogether (since it's kind of redundant).

CT075 commented on bin/connector_config/Generate.hs:
```diff
@@ -37,38 +37,66 @@ data HolyWeapon
         | i < 1 = error $ "invalid chapter index " ++ show i
         | i < 6 = C i
         | i < 22 = C (i - 1)
+        | i < 32 = T (i)
```
This is not really how this function works.

The goal of this function is to map a number from 0-[however many chapters we have] to that chapter following some ordering. So 0 -> Prologue, 1 -> Chapter 1, etc. Right now, you're mapping the number 23 to "Tower 23", which is not a real location. What we actually want to do is line them up like

Prologue
C 1
...
C5x
C 6
...
T 1
T 2
...
R 1
...
R 10

and then do some logic to map the numbers properly.

Darrman committed:
Rearranged chapter numbers

Victory is now the final location again as per review, and the hardcoded Ruins
10 has been de-hardcoded. Relevant enums appear to work correctly.
Still doesn't build; item placement is broken.

Darrman committed:
Items present in base patch

Still testing to make sure things work.

CT075 reviewed (changes requested):
Mostly looks good, but there are still some issues with `Generate.hs` (see the other PR).

CT075 committed:
minor comment changes

Signed-off-by: Cameron Wong <cam@camdar.io>

CT075 committed:
Merge upstream 'main' into filleritems

Signed-off-by: Cameron Wong <cam@camdar.io>
